src/core_plugins/kibana/public/dashboard/dashboard_state_manager.js

```diff
@@ -582,6 +588,7 @@ export class DashboardStateManager {
     this.savedDashboard.searchSource.setField('query', query);
     this.savedDashboard.searchSource.setField('filter', filters);
     this.saveState();
+    this._pushFiltersToStore();
```

Comment:
strange. If I comment out this function entirely and edit a filter pill, the URL and the UI still get updated.  I wonder if we even need all this code here.  Although maybe we should just go for the quick fix and file this as more tech debt to look at later.

Reply:
I think the meat of this exploration needs to get pushed till later.  For now, this works, so LGTM.  I do think it could go for a comment though, e.g. `pinned filters go on global state, therefore are not propagated to store via app state and have to be pushed manually.`

This also makes me realize that the dashboard redux store filters have no distinction between global vs app state (unless I guess you look at their internals).  I don't think this should cause any problems **right now**, but I can easily see this becoming a bug in the future.  We push all filters (global and app) to the store. If the store filters get edited, and we want to push those changes to the URL, we need to know which ones go into global and which ones app.

But right now, in `_handleStoreChanges` we don't deal with filters at all (well, only staged filters, so I think this is fine).

tl;dr good for now but this is fragile code that I imagine will come back to bite us (not your changes, but the way it currently works).

Reply:
yeah, we were ignoring pinned filters till now when copying to redux state ....